iotedgedev/cli.py

```diff
@@ -341,12 +341,16 @@ def setup_simulator(gateway_host, iothub_connection_string):
               default=53000,
               show_default=True,
               help="Port of the service for sending message.")
+@click.option("--iothub-connection-string",
+              "-c",
+              help="Set Azure IoT Hub connection string when setup IoT Edge simulator. Note: Use double quotes when supplying this input.",
+              required=False)
 @with_telemetry
-def start_simulator(setup, solution, build, manifest_file, platform, verbose, inputs, port):
+def start_simulator(setup, solution, build, manifest_file, platform, verbose, inputs, port, iothub_connection_string):
     sim = Simulator(envvars, output)
 
     if setup:
-        sim.setup(socket.getfqdn())
+        sim.setup(socket.getfqdn(), iothub_connection_string)
```

Comment:
is there any reason must provide hub connection string? it should not fail if not provided.

Reply:
Previously iothub_connection_string is been implemented as a required parameter, so not passing this parameter will cause python runtime error. If passing an empty string to it, iotedgedev won't append iothub connection string parameter when setting up iotedgehubdev.
I have make this parameter optional.